wcivf/apps/elections/models.py

```diff
@@ -417,12 +400,29 @@ class PostElection(TimeStampedModel):
         choices=ElectionCancellationReason.choices,
         default=None,
     )
+    ballot_papers_issued = models.IntegerField(blank=True, null=True)
+    electorate = models.IntegerField(blank=True, null=True)
+    turnout = models.IntegerField(blank=True, null=True)
+    spoilt_ballots = models.IntegerField(blank=True, null=True)
 
     objects = PostElectionQuerySet.as_manager()
 
     class Meta:
         get_latest_by = "ynr_modified"
 
+    @property
+    def has_results(self):
+        """
+        Returns a boolean for if the election has results
+        """
+        return bool(
+            self.spoilt_ballots
+            or self.ballot_papers_issued
+            or self.turnout
+            or self.electorate
+            or self.postelection_set.filter(personpost__elected=True).exists()
```

Comment:
Given that this is on the `PostElection` model, I don't think `self.postelection_set` is referring to anything. An `Election` object had a `postelection_set` because there's a many-to-one (aka foreign key) relationship from `PostElection` to `Election`. 

TLDR: I think this line should be `self.personpost_set.filter(elected=True)`.

More broadly this property is a bit confusing to me because there seems to be 5 different things that are sufficient to show results - is it the case that we want to show results when any one of these is true? have you checked that this makes sense on the front end (don't necessarily need to write tests for all outcomes, more just sanity checking it.

Reply:
This is partly my fault...

The property is called `has_results`, so I was worried that future us might use it to, e.g change a template to indicate that the election was over (or something). In other words, `has_results` suggests some sort of interface contract.

Seeing as the first 4 things are optional values that can all be `None`, there could be a situation where we _do_ have results, but just no values in these fields. 

This is why I suggested adding the final check: if someone was elected then we surely have results, and if someone wasn't then we surely don't. The previous checks are just an early exit system to save a query.

Reply:
Changed with https://github.com/DemocracyClub/WhoCanIVoteFor/pull/1720/commits/c3539742fef29cebb258c263e5f6bfe66f5114a0

Reply:
We don't show the table item if the cell is blank.